src/main/java/com/github/pedrovgs/androidwifiadb/view/CardLayoutDevices.java

```diff
@@ -0,0 +1,144 @@
+package com.github.pedrovgs.androidwifiadb.view;
+
+import com.github.pedrovgs.androidwifiadb.model.Device;
+import com.intellij.ui.table.JBTable;
+
+import java.awt.BorderLayout;
+import java.awt.CardLayout;
+import java.awt.Container;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.JCheckBox;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTable;
+import javax.swing.SwingConstants;
+
+/**
+ * Created by vgaidarji on 10/26/15.
+ */
+public class CardLayoutDevices implements ActionButtonListener {
+
+    private static final String CARD_DEVICES = "Card with JTable devices";
+    private static final String CARD_NO_DEVICES = "Card with no devices info";
+
+    private Container parentContainer;
+    private JPanel cards;
+    private JPanel panelNoDevices;
+    private JPanel panelDevices;
+    private JTable tableDevices;
+    private List<Device> devices;
+    private DeviceAction deviceAction;
+
+    public CardLayoutDevices(Container parentContainer, DeviceAction action) {
+        this.deviceAction = action;
+        this.parentContainer = parentContainer;
+        this.devices = new ArrayList<Device>();
+    }
+
+    public void setDevices(List<Device> devices) {
+        this.devices = devices;
+    }
+
+    /**
+     * Create and show Devices UI in parentContainer.
+     */
+    public void createAndShowGUI() {
+        cards = new JPanel(new CardLayout());
+        createNoDevicesPanel();
+        createTableDevices();
+        cards.add(panelDevices, CARD_DEVICES);
+        cards.add(panelNoDevices, CARD_NO_DEVICES);
+        parentContainer.add(cards, BorderLayout.PAGE_START);
+        setupUi();
+    }
+
+    private void setupUi() {
+        if (devices != null && devices.size() > 0) {
+            showCard(CARD_DEVICES);
+        } else {
+            showCard(CARD_NO_DEVICES);
+        }
+    }
+
+    /**
+     * Shows appropriate card, depending on connected devices.
+     */
+    public void updateUi() {
+        if (devices != null && devices.size() > 0) {
```

Comment:
We can replace ``devices.size() > 0`` with ``!devices.isEmpty()``. Review all the usages you have of ``size() > 0`` and use ``isEmpty()`` instead.